ravenpy/models/emulators.py

```diff
@@ -640,6 +646,158 @@ def ost2raven(self, ops):
         out = [ops[n] for n in names]
         return self.params(*out)
 
+class SACSMA(Raven):
+
+    # Details about SAC-SMA:
+    # https://wiki.ewater.org.au/display/SD41/Sacramento+Model+-+SRG
+
+    identifier = "sacsma"
+    templates = tuple((Path(__file__).parent / "raven-sacsma").glob("*.rv?"))
+
+    params = namedtuple(
+        "SACSMAParams", ", ".join(["par_x{:02}".format(i) for i in range(1, 22)])
+    )
+
+    @dataclass
+    class HRU(HRU):
+        land_use_class: str = "FOREST"
+        veg_class: str = "FOREST"
+        soil_profile: str = "DEFAULT_P"
+        aquifer_profile: str = "[NONE]"
+        terrain_class: str = "[NONE]"
+
+    def __init__(self, *args, **kwds):
+        super().__init__(*args, **kwds)
+        self.rvp = RVP(params=SACSMA.params(*((None,) * len(SACSMA.params._fields))),
+                       land_use_classes=(LU("FOREST", impermeable_frac=0.0, forest_coverage=0.02345),)
+                       )
+        self.rvh = RVH(hrus=(SACSMA.HRU(),))
+        self.rvt = RVT(**{k: nc() for k in std_vars})
+        self.rvi = RVI(evaporation="PET_OUDIN", rain_snow_fraction="RAINSNOW_DATA")  # would like to use RAINSNOW_HBV??
+        self.rvc = RVC(soil0=None, soil2=None, basin_state=BasinIndexCommand())
+        self.rvd = RV(
+            par_soil0_mm=None,
+            par_soil2_mm=None,
+            PAR_BF_LOSS_FRAC=None,
+            POW_X01=None,
+            POW_X02=None,
+            POW_X03=None,
+        )
+
+    def derived_parameters(self):
+        self.rvd["par_soil0_mm"] = self.rvp.params.par_x04 * 1000.0
+        self.rvd["par_soil2_mm"] = self.rvp.params.par_x06 * 1000.0
+        self.rvd[
+            "PAR_BF_LOSS_FRAC"
+        ] = self.rvp.params.par_x12  # PAR_BF_LOSS_FRAC = par_x12/(1+par_x12)
```

Comment:
Please put your comments on the previous line (not at the end of the line) so that Black does not do this strange formatting (due to line lenght being > 88). Also if you are not already doing it, please use `pre-commit`, in your venv:
```
pip install pre-commit # if not already installed in your venv
pre-commit install
```
This will run some formatting and verification tools on your source code (black, isort and flake8) just before the commit stage, and it will prevent the commit from happening if it cannot resolve all the problems by itself.

Reply:
Done.